tests/unit/Option/NoneTest.php

```diff
@@ -130,4 +130,44 @@ public function testEquality(): void
         static::assertTrue($a->equals(Option\none()));
         static::assertFalse($a->equals(Option\some('other')));
     }
+
+    public function testZip(): void
+    {
+        $x = Option\some(1);
+        $y = Option\none();
+
+        static::assertTrue($x->zip($y)->isNone());
+        static::assertTrue($y->zip($x)->isNone());
+    }
+
+    public function testZipWith(): void
+    {
+        $x = Option\some(1);
+        $y = Option\none();
+
+        static::assertTrue($x->zipWith($y, static fn($a, $b) => $a + $b)->isNone());
+        static::assertTrue($y->zipWith($x, static fn($a, $b) => $a + $b)->isNone());
+    }
+
+    /**
+     * @dataProvider provideTestUnzip
+     */
+    public function testUnzip(Option\Option $option): void
+    {
+        [$x, $y] = $option->unzip();
```

Comment:
If you implement the change requested above, you'll only need to validate the none() case.